src/main/java/bio/terra/cli/command/resource/update/GcpDataprocCluster.java

```diff
@@ -0,0 +1,92 @@
+package bio.terra.cli.command.resource.update;
+
+import bio.terra.cli.businessobject.Context;
+import bio.terra.cli.businessobject.Resource.Type;
+import bio.terra.cli.command.shared.WsmBaseCommand;
+import bio.terra.cli.command.shared.options.Format;
+import bio.terra.cli.command.shared.options.ResourceUpdate;
+import bio.terra.cli.command.shared.options.WorkspaceOverride;
+import bio.terra.cli.exception.UserActionableException;
+import bio.terra.cli.serialization.userfacing.input.UpdateControlledGcpDataprocClusterParams;
+import bio.terra.cli.serialization.userfacing.resource.UFGcpDataprocCluster;
+import picocli.CommandLine;
+
+@CommandLine.Command(
+    name = "dataproc-cluster",
+    description = "Update a GCP Dataproc cluster.",
+    showDefaultValues = true)
+public class GcpDataprocCluster extends WsmBaseCommand {
+  @CommandLine.Mixin ResourceUpdate resourceUpdateOptions;
+  @CommandLine.Mixin WorkspaceOverride workspaceOption;
+  @CommandLine.Mixin Format formatOption;
+
+  @CommandLine.Option(names = "--num-workers", description = "The number of worker nodes.")
+  private Integer numWorkers;
+
+  @CommandLine.Option(
+      names = "--num-secondary-workers",
+      description = "The number of secondary worker nodes.")
+  private Integer numSecondaryWorkers;
+
+  @CommandLine.Option(
+      names = "--autoscaling-policy",
+      description =
+          "Autoscaling policy url to attach to the cluster. Format: projects/[projectId]/locations/[dataproc_region]/autoscalingPolicies/[policy_id]")
+  private String autoscalingPolicy;
+
+  @CommandLine.Option(
+      names = "--graceful-decommission-timeout",
+      description = "The duration to wait for graceful decommissioning to finish.")
+  private String gracefulDecommissionTimeout;
+
+  @CommandLine.Option(
+      names = "--idle-delete-ttl",
+      description = "Time-to-live after which the resource becomes idle and is deleted.")
+  public String idleDeleteTtl;
+
+  /** Print this command's output in text format. */
+  private static void printText(UFGcpDataprocCluster returnValue) {
+    OUT.println("Successfully updated GCP Dataproc cluster.");
+    returnValue.print();
+  }
+
+  /** Update a GCP Dataproc cluster in the workspace. */
+  @Override
+  protected void execute() {
+    workspaceOption.overrideIfSpecified();
+
+    // all update parameters are optional, but make sure at least one is specified
+    if (!resourceUpdateOptions.isDefined()
```

Comment:
I think that would be nicer, but I need at least one between these update parameters AND resource name, description in `resourceUpdateOptions`, but picocli doesn't support mixins inside arg groups. Also this would change the behavior a little where the command `--help` would show the args in a subsection.